hathor/transaction/storage/transaction_storage.py

```diff
@@ -584,7 +625,15 @@ def get_all_tips(self, timestamp: Optional[float] = None) -> Set[Interval]:
         assert self.all_index is not None
         if timestamp is None:
             timestamp = self.latest_timestamp
+
+        if self._all_tips_cache is not None and timestamp >= self._all_tips_cache.timestamp:
+            return self._all_tips_cache.tips
+
         tips = self.all_index.tips_index[timestamp]
+        if timestamp >= self.latest_timestamp:
```

Comment:
This cache invalidation mechanism seems to fail if a new transaction arrives with the same timestamp as the previous one (or with an older timestamp). In this case, I think we will wrongly return an outdated tips list.

Reply:
Not really. In `_add_to_cache`, the tips cache is always invalidated, regardless of the received transaction's timestamp. That's where the cache is invalidated.

Reply:
Yes, exactly. If a new transaction arrives in an previous (or older) timestamp, the cache will be invalidated when adding it to the cache and the tips will be recalculated when this method is executed again with a `timestamp >= latest_timestamp`

Reply:
I finally got it. I haven't seen the change in the `_add_to_cache` method. You're right! Thanks!